ZAPD/ZFile.cpp

```diff
@@ -815,12 +592,12 @@ void ZFile::GenerateHLIntermediette()
 
 	for (ZResource* res : resources)
 	{
-		if (typeid(ZDisplayList) == typeid(*res) || typeid(ZSkeleton) == typeid(*res))
+		if (res->GetResourceType() == ZResourceType::DisplayList || res->GetResourceType() == ZResourceType::Skeleton)
```

Comment:
I think it may be a better idea to add a `HasHLIntermediette()` method in `ZResource`, and the resources that implement that interface simply return true in that method. Instead of having to hardcoded which resources implement this.